source/PlayerInfo.cpp

```diff
@@ -1668,20 +1668,23 @@ bool PlayerInfo::TakeOff(UI *ui)
 		RemoveMission(Mission::ABORT, *mission, ui);
 
 	// Any ordinary cargo left behind can be sold.
-	int64_t income = 0;
+	int64_t commoditiesIncome = 0, outfitsIncome = 0;
 	int day = date.DaysSinceEpoch();
-	int64_t sold = cargo.Used();
+	int64_t commoditiesSold = cargo.Used() - cargo.OutfitsSize();
+	int64_t outfitsSold = cargo.Used() - cargo.CommoditiesSize();
 	int64_t totalBasis = 0;
-	if(sold)
+	if(commoditiesSold)
 	{
 		for(const auto &commodity : cargo.Commodities())
 		{
 			if(!commodity.second)
 				continue;
 
 			// Figure out how much income you get for selling this cargo.
-			int64_t value = commodity.second * static_cast<int64_t>(system->Trade(commodity.first));
-			income += value;
+			int64_t value = 0;
+			if(hasSpaceport)
```

Comment:
```suggestion
			if(planet->IsInhabited())
```

Reply:
Given that hasSpaceport == HasSpaceport() && CanUseServices() I think it should be rather
```suggestion
			if(planet->IsInhabited() && planet->CanUseServices())
```
(btw can't you move this condition up so that if there's no trade on the planet, the whole loop doesn't get executed?)

Reply:
Indeed, I've pushed changes that should resolve this.